modules/basics/src/main/java/com/opengamma/strata/basics/currency/FxConvertible.java

```diff
@@ -35,4 +37,20 @@ public interface FxConvertible<R> {
    */
   public abstract R convertedTo(Currency resultCurrency, FxRateProvider rateProvider);
 
+  /**
+   * Converts this instance to an equivalent amount in the specified currency.
+   * <p>
+   * The result, which may be of a different type, will be expressed in terms of the given currency.
+   * Any FX conversion that is required will use rates from the provider.
+   * <p>
+   * If no FX rate is found the returned result will be a failure.
+   *
+   * @param resultCurrency  the currency of the result
+   * @param rateProvider  the provider of FX rates
+   * @return the result of the converted instance, which should be expressed in the specified currency
+   */
+  public default Result<R> convertedToResult(Currency resultCurrency, FxRateProvider rateProvider) {
+    return Result.of(() -> convertedTo(resultCurrency, rateProvider));
```

Comment:
Given this implementation, I'm struggling to justify adding the method,. Bear in mind that the method becomes public on a large number of implementation classes.

Reply:
We have `Result.of(() -> ...convertedTo)` littered around most of the models, as we want to avoid any hard errors in a calculation, this method would help clean things up a lot.